core/deployment/src/main/java/io/quarkus/deployment/shutdown/ShutdownBuildTimeConfig.java

```diff
@@ -0,0 +1,18 @@
+package io.quarkus.deployment.shutdown;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public class ShutdownBuildTimeConfig {
+
+    /**
+     * Delay between shutdown being requested and actually initiated. Also called the pre-shutdown phase.
+     * In pre-shutdown, the server continues working as usual, except a readiness probe starts reporting "down"
+     * (if the {@code smallrye-health} extension is present). This gives the infrastructure time to detect
+     * that the application instance is shutting down and stop routing traffic to it.
+     */
```

Comment:
> Whether Quarkus should wait between shutdown being requested and actually initiated.
> This delay gives the infrastructure time to detect that the application instance is shutting down and stop routing traffic to it.

Reply:
Or something like that.

Reply:
hmm... :smile: 
https://github.com/quarkusio/quarkus/pull/39206#discussion_r1513991091

Reply:
This is the new build-time config property, `delay-enabled`.

Reply:
Done